sound/soc/sof/amd/acp.c

```diff
@@ -138,25 +138,40 @@ int configure_and_run_dma(struct acp_dev_data *adata, unsigned int src_addr,
 	return ret;
 }
 
-static int psp_fw_validate(struct acp_dev_data *adata)
+static int psp_mbox_ready(struct acp_dev_data *adata)
 {
 	struct snd_sof_dev *sdev = adata->dev;
 	int timeout;
 	u32 data;
 
-	smn_write(adata->smn_dev, MP0_C2PMSG_26_REG, MBOX_ACP_SHA_DMA_COMMAND);
-
 	for (timeout = ACP_PSP_TIMEOUT_COUNTER; timeout > 0; timeout--) {
 		msleep(20);
 		smn_read(adata->smn_dev, MP0_C2PMSG_26_REG, &data);
 		if (data & MBOX_READY_MASK)
 			return 0;
 	}
 
-	dev_err(sdev->dev, "FW validation timedout: status %x\n", data & MBOX_STATUS_MASK);
+	dev_dbg(sdev->dev, "PSP error status %x\n", data & MBOX_STATUS_MASK);
```

Comment:
how is this not an error?

Reply:
dev_err msg for this failure is already added in ln#216 (https://github.com/thesofproject/linux/pull/3496/commits/787ce20d75e4c26c32023eda7d20b8b9732e708b#diff-e710ad418c0e2541fd93217c274e2ec3d05fa33f41aa9eeb6c232bc47122925bR216)

So changed this to dev_dbg() as this status dump needed for debugging only in case of send command failure

Reply:
that's not intuitive. an error has to remain an error and it should be printed exactly once and its better to be printed in the function that returns the original error so that all calers will see it.

Reply:
Ok .. will changed this to dev_err() and remove it from caller